packages/modular-scripts/src/rename.ts

```diff
@@ -0,0 +1,107 @@
+import { Project } from 'ts-morph';
+import execa from 'execa';
+import actionPreflightCheck from './utils/actionPreflightCheck';
+import getModularRoot from './utils/getModularRoot';
+import getWorkspaceInfo from './utils/getWorkspaceInfo';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as logger from './utils/logger';
+import { cleanGit, stashChanges } from './utils/gitActions';
+
+process.on('SIGINT', () => {
+  stashChanges();
+  process.exit(1);
+});
+
+// Derive types from array; this is handy to iterate against it later
+const dependencyTypes = [
+  'dependencies',
+  'devDependencies',
+  'peerDependencies',
+  'optionalDependencies',
+] as const;
+
+type DependencyType = typeof dependencyTypes[number];
+
+type DependencyObject = {
+  [key in DependencyType]: Record<string, string>;
+};
+
+interface PackageJson extends Partial<DependencyObject> {
+  name: string;
+}
```

Comment:
You should also be able to get those dependencty types from `import { CoreProperties } from '@schemastore/package'`

Reply:
Good catch, this was handy before, where I iterated on the depTypes const array and constructed the types from it, but it's not needed now. Done.